app/src/components/App.tsx

```diff
@@ -1,38 +1,28 @@
 import * as React from 'react';
 
-import { ContractProps } from '../helper/eth';
-
-import Web3 from 'web3';
 import { Web3Provider } from '../providers/Web3Provider';
 import { NameServiceProvider } from '../providers/NameServiceProvider';
 import { RegistryProvider } from '../providers/RegistryProvider';
 import { SessionProvider } from '../providers/SessionProvider';
+import { ContractProvider } from '../providers/ContractProvider';
+import { ArtistProvider } from '../providers/ArtistProvider';
 
 import Router from './Router';
 
-export interface AppProps extends ContractProps {
-  web3: Web3;
-  networkId: number;
-}
-
-const ContractProvider: React.FC = ({ children }) => {
-  return (
-    <NameServiceProvider>
-      <RegistryProvider>
-        {children}
-      </RegistryProvider>
-    </NameServiceProvider>
-  );
-};
-
-const App: React.FC<AppProps> = (props) => {
+const App: React.FC<{}> = () => {
   return (
-    <Web3Provider networkId={props.networkId} accounts={props.accounts}>
-      <ContractProvider>
-        <SessionProvider address={props.accounts[0]} contracts={props.contracts}>
-          <Router {...props}/>
-        </SessionProvider>
-      </ContractProvider>
+    <Web3Provider>
+      <NameServiceProvider>
+        <ContractProvider>
+          <ArtistProvider>
+            <RegistryProvider>
```

Comment:
Registry is a Contract?

Reply:
Yeah this was something that was done before this PR. For somereason there was a RegistryProvider which just connects you to the contract even though we were also grabbing all contracts in index and passing those down. I imagine the provider option was to reduce the passing down of contracts. I agree something needs to be fixed and am happy to do that but would rather do in a seperate PR?